admin_guide/iptables.adoc

```diff
@@ -11,18 +11,89 @@
+the kernel iptables configuration for proper network operation. In addition the iptables
+configuration of all nodes in the cluster must be correct for networking to work.
 
+All components independently work with iptables without knowledge of how other components
+are using them. This makes it very easy for one component to break another component's configuration.
+Further, {product-title} and Docker assume that iptables remains set up exactly as they have set it up.
+They do not detect problems introduced by other components.
 
-== Restarting
-Docker doesn't monitor the iptables rules that it adds for exposing ports from containers and hence if
-iptables service is restarted, then these rules are lost. So, to safely restart iptables, it is
-recommended that the rules are saved and restored. 
+IMPORTANT: You must make sure that whatever changes you make to the iptables configuration on a node
+do not impact the operation of {product-title} and Docker. Also, changes will frequently need to be made on
+all nodes in the cluster. Be careful, the setup is very complex and it is very easy to break {product-title}
+and Docker networking.
+
+The chains, order of the chains and rules in the kernel iptables must be properly set up on each node
+in the cluster for {product-title} and docker networking to work properly. There are several tools and
+services that are commonly used in the system that interact with the kernel iptables and can
+accidentally impact {product-title} and Docker.
+
+=== iptables
+Iptables is a low level tool that can be used to set up, maintain, and inspect the tables of IPv4
+packet filter rules in the Linux kernel.
+
+WARNING: This is a very powerful tool that can manage all aspects of the kernel iptables. It assumes
+you know what you are doing.
+
+Independent of other use, such as a firewall, {product-title} and Docker manage chains in some of the tables.
+The chains are inserted in specific order and the rules are specific to their needs.
+
+IMPORTANT: iptables --flush [chain] can remove key required configuration. Be very careful when using this.
+
+When making changes consider limiting your changes to the OS_FIREWALL_ALLOW chain in the filter table.
+
+
+=== firewalld
+
+When you need to set up a firewall on the cluster's nodes, use firewalld.
+Firewalld can be used to safely manage access to the cluster in a way that
+does not confiict with {product-title} or Docker. For example, when a
+
+----
+# firewall-cmd --reload
+----
+
+command is executed both Openshift and Docker are notified so they can restore any needed rules.
+
+Detailed instructions for working with firewalld are beyond the scope of this document.
+
+
+=== iptables.service
+The iptables service supports a network firewall. It assumes total control of the iptables configuration
+and it flushes and restores the complete iptables configuration. It is not integrated with either Docker
+or {product-title} so when it is started it breaks cluster networking.
```

Comment:
when it is started or reloaded it restores a saved snapshot of the firewall rules and usually breaks cluster networking.